util/internal/integrationtest/common/src/main/java/org/hibernate/search/util/impl/integrationtest/common/stub/backend/types/impl/StubIndexValueFieldType.java

```diff
@@ -61,7 +61,7 @@ public void apply(StubIndexSchemaDataNode.Builder builder) {
 		public Builder(Class<F> valueClass) {
 			super( valueClass );
 			stubFactories(
-					new StubSearchPredicate.Factory(),
+					new StubSearchPredicate.RegexpFactory(),
 					PredicateTypeKeys.NESTED,
 					PredicateTypeKeys.MATCH,
 					PredicateTypeKeys.RANGE,
```

Comment:
You're assigning the regexp predicate factory to all sorts of predicates that are not the regexp predicate.

The right way to do what you want would be to remove PredicateTypeKeys.REGEXP from the list, and add a call to `stubFactories(new StubSearchPredicate.RegexpFactory(), PredicateTypeKeys.REGEXP)`.

Though TBH, this looks like dead code, so it's more for consistency than anything else.